pages/upload.py

```
@@ -57,99 +58,11 @@ def analyze_and_process_excel(uploaded_file, file_type="Auto-detectar"):
-                    if new_col not in existing_columns or old_col == new_col:
-                        if new_col not in target_columns:
-                            safe_rename[old_col] = new_col
-                            target_columns.add(new_col)
-                        else:
-                            # Skip this mapping to avoid duplicate
-                            st.warning(f"⚠️ Skipping duplicate mapping: '{old_col}' → '{new_col}'")
-            
-            # Apply the safe renaming
-            df_full = df_full.rename(columns=safe_rename)
+            df_full, change_pairs = apply_column_remap(df_full)
             renamed_columns = list(df_full.columns)
```

Flake8 (7.2.0)

[error] local variable 'renamed_columns' is assigned to but never used

(F841)